galaxy_ng/tests/integration/dab/test_dab_rbac_contract.py

```diff
@@ -111,11 +117,56 @@ def delete_role():
 @pytest.fixture
 def namespace(galaxy_client):
     gc = galaxy_client("admin")
-    payload = {"name": "new_namespace"}
-    ns = gc.post("_ui/v1/my-namespaces/", body=payload)
-    yield ns
-    with pytest.raises(ValueError):
-        gc.delete(f"_ui/v1/my-namespaces/{ns['name']}/")
+
+    namespace = gc.post("_ui/v1/my-namespaces/", body={"name": "new_namespace"})
+    yield namespace
+
+    response = gc.delete(f"_ui/v1/my-namespaces/{namespace['name']}/", parse_json=False)
+    assert response.status_code == HTTPStatus.NO_CONTENT
+
+
+# TODO(cutwater): Refactor these assert functions.
+#   Maybe split into separate assertions. Check against expected payload.
+# NOTE(cutwater): The goal is to check that assignment is duplicated in both systems.
+def assert_object_role_assignments(gc, user, namespace, expected=0):
+    # Assure the assignment shows up in the pulp API
+    data = gc.get(f"_ui/v1/my-namespaces/{namespace['name']}/")
+    assert len(data["users"]) == expected
+
+    # Assure the assignment shows up in the DAB RBAC API
+    data = gc.get(f"_ui/v2/role_user_assignments/?user={user['id']}&object_id={namespace['id']}")
+    assert data["count"] == expected
+
+    if not expected:
+        return
+
+    # Ensure summary_fields is populated with expected sub keys
+    summary_fields = data["results"][0]["summary_fields"]
+    expected_fields = {"created_by", "role_definition", "user", "content_objet"}
+    assert expected_fields.issubset(summary_fields)
+    # assert each entry has at least the id field
+    for field in expected_fields:
+        assert "id" in summary_fields[field]
+
+
+def assert_system_role_assignments(client, user, expected=0):
```

Comment:
The other method `assert_object_role_assignments` takes `namespace` as an argument. It _could_ also take the role name as an argument.

For this method, there's no object (like namespace), but these args leave the method as too general. It would count _any_ role the user has, which is prone to false-positives.

Upshot: I think this method should take the _role name_ and include that in filters.